### ISSUE TITLE
Host ip can not connect to the following ports [ip:2379]

### ISSUE THREAD

**Issue by Aisuko** [rancher/rke#955]
I was deploy succeed with these steps below oncetime,  please help me with this issue,thanks.

**RKE version:**
rke version v0.1.10

**Docker version: (`docker version`,`docker info` preferred)**
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:31 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:29:02 2018
  OS/Arch:          linux/amd64
  Experimental:     true

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

* kernel

4.4.0-1067-aws

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

AWS

**cluster.yml file:**

```
nodes:
- address: aws ip
  port: "22"
  internal_address: ""
  role:
  - controlplane
  - worker
  - etcd
  hostname_override: ""
  user: ubuntu
  docker_socket: /var/run/docker.sock
  ssh_key: ""
  ssh_key_path: /test_rke.pem
  labels: {}
services:
  etcd:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
    snapshot: false
    retention: ""
    creation: ""
  kube-api:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    service_cluster_ip_range: 10.43.0.0/16
    service_node_port_range: ""
    pod_security_policy: false
  kube-controller:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
  scheduler:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
  kubelet:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    cluster_domain: cluster.local
    infra_container_image: ""
    cluster_dns_server: 10.43.0.10
    fail_swap_on: false
  kubeproxy:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
network:
  plugin: canal
  options: {}
authentication:
  strategy: x509
  options: {}
  sans: []
addons: ""
addons_include: []
system_images:
  etcd: rancher/coreos-etcd:v3.2.18
  alpine: rancher/rke-tools:v0.1.15
  nginx_proxy: rancher/rke-tools:v0.1.15
  cert_downloader: rancher/rke-tools:v0.1.15
  kubernetes_services_sidecar: rancher/rke-tools:v0.1.15
  kubedns: rancher/k8s-dns-kube-dns-amd64:1.14.10
  dnsmasq: rancher/k8s-dns-dnsmasq-nanny-amd64:1.14.10
  kubedns_sidecar: rancher/k8s-dns-sidecar-amd64:1.14.10
  kubedns_autoscaler: rancher/cluster-proportional-autoscaler-amd64:1.0.0
  kubernetes: rancher/hyperkube:v1.11.3-rancher1
  flannel: rancher/coreos-flannel:v0.10.0
  flannel_cni: rancher/coreos-flannel-cni:v0.3.0
  calico_node: rancher/calico-node:v3.1.3
  calico_cni: rancher/calico-cni:v3.1.3
  calico_controllers: ""
  calico_ctl: rancher/calico-ctl:v2.0.0
  canal_node: rancher/calico-node:v3.1.3
  canal_cni: rancher/calico-cni:v3.1.3
  canal_flannel: rancher/coreos-flannel:v0.10.0
  wave_node: weaveworks/weave-kube:2.1.2
  weave_cni: weaveworks/weave-npc:2.1.2
  pod_infra_container: rancher/pause-amd64:3.1
  ingress: rancher/nginx-ingress-controller:0.16.2-rancher1
  ingress_backend: rancher/nginx-ingress-controller-defaultbackend:1.4
  metrics_server: rancher/metrics-server-amd64:v0.2.1
ssh_key_path: ~/.ssh/id_rsa
ssh_agent_auth: false
authorization:
  mode: rbac
  options: {}
ignore_docker_version: false
kubernetes_version: ""
private_registries: []
ingress:
  provider: ""
  options: {}
  node_selector: {}
  extra_args: {}
cluster_name: ""
cloud_provider:
  name: ""
prefix_path: ""
addon_job_timeout: 0
bastion_host:
  address: ""
  port: ""
  user: ""
  ssh_key: ""
  ssh_key_path: ""
monitoring:
  provider: ""
  options: {}```

**Steps to Reproduce:**
```
➜  rke up --config cluster.yml
 ```

**Results:**

```
INFO[0000] Building Kubernetes cluster                  
INFO[0000] [dialer] Setup tunnel for host [aws ip] 
INFO[0002] [network] Deploying port listener containers 
INFO[0006] [network] Port listener containers deployed successfully 
INFO[0006] [network] Running control plane -> etcd port checks 
INFO[0008] [network] Successfully started [rke-port-checker] container on host [aws ip] 
FATA[0013] [network] Host [aws ip] is not able to connect to the following ports: [aws ip:2379]. Please check network policies and firewall rules
```

**Comment by superseb** [comment]
If the nodes are newly created (or cleaned properly), its either security group (public<->private) or host firewall blocking the ports. Let me know if you still need help on this.

**Comment by dunatotatos** [comment]
Same issue here. Things I've tried without success:

- disabling the firewall
- disabling IPv6 (the hostname resolves as an IPv4 address)

However, changing the node IP address from hostname to the IP solved the issue, even if the DNS resolution works as expected... I don't know why.

**Comment by luis13byte** [comment]
I got this error in a network with more nodes and without firewall. In my case the solution was the following:

- Ensure that in ALL control planes can resolve the hostname of the new worker (I used a local host).

- That the server that is going to join the cluster can resolve the hostnames of all the control planes.

Therefore the error as such was not because of the port, as indicated, but because the servers could not reach each other.
